File: src/jk_5/nailed/command/CommandSpectator.java
package jk_5.nailed.command;

import jk_5.nailed.Nailed;
import jk_5.nailed.players.Player;
import net.minecraft.src.CommandBase;
import net.minecraft.src.CommandException;
import net.minecraft.src.EntityPlayerMP;
import net.minecraft.src.ICommandSender;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandSpectator extends CommandBase {

    @Override
    public String getCommandName() {
        return "spectator";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 0;
    }

    @Override
    public boolean canCommandSenderUseCommand(ICommandSender sender) {
        return sender instanceof EntityPlayerMP;
    }

    @Override
    public String getCommandUsage(ICommandSender par1ICommandSender) {
        return "/spectator - Toggles spectator mode";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        Player p = Nailed.playerRegistry.getPlayer(sender.getCommandSenderName());
        if (p == null) throw new CommandException("I seriously don\'t know who you are!");
        p.setSpectator(!p.isSpectator());
    }
}

File: src/jk_5/nailed/command/CommandSpectator.scala
package jk_5.nailed.command

/**
 * No description given
 *
 * @author jk-5
 */
object CommandSpectator extends TCommand {
  val commandName = "spectator"

  override def canCommandSenderUseCommand(sender: CommandSender) = sender.isValidPlayer
  override def getCommandUsage = "/spectator - Toggles spectator mode"
  def processCommand(sender: CommandSender, args: Array[String]){
    if(sender.isInvalidPlayer) throw new CommandException("I seriously don\'t know who you are!")
    if(!sender.map.getGameThread.isGameRunning) throw new CommandException("You can\'t join spectator mode when no game is running")
    sender.player.setSpectator(sender.player.isSpectator)
  }
}
